kong/plugins/http-log/handler.lua

```diff
@@ -189,12 +193,43 @@ function HttpLogHandler:log(conf)
 
   local queue_conf = Queue.get_plugin_params("http-log", conf, make_queue_name(conf))
   kong.log.debug("Queue name automatically configured based on configuration parameters to: ", queue_conf.name)
+  local entry = cjson.encode(kong.log.serialize())
+
+  if queue_conf.max_batch_size == 1 then
+    local ok ,err = timer_at(0, function()
+      local retry_count = 0
+      local start_time = now()
+      while true do
+        local ok, err = send_entries(conf, { entry })
+        if ok then
+          break
+        end
+        if not ok then
+          kong.log.warn(fmt("handler could not process entries: %s",
+            tostring(err or "no error details returned by handler")))
+        end
+
+        if (now() - start_time) > queue_conf.max_retry_time then
+          kong.log.err("could not send entries due to max_retry_time exceeded. 1 queue entry was lost")
```

Comment:
```suggestion
          kong.log.err("could not send this log entry due to max_retry_time exceeded: lost")
```

Reply:
We now reuse the `Queue.handle` logic, so the suggestion here is not applicable.